zooniverse/front-end-monorepo
# lib-user: Add certificate page

## Package
- app-root
- lib-react-components
- lib-user

## Describe your changes
Figma:
- pages: https://www.figma.com/design/qbqbmR3t5XV6eKcpuRj7mG/Group-Stats?node-id=16-110&t=evrDTlBn4g63ZzlS-4
- certificates: https://www.figma.com/design/qbqbmR3t5XV6eKcpuRj7mG/Group-Stats?node-id=1827-6388&t=evrDTlBn4g63ZzlS-4

Packages:
- app-root
  - add certificate page
  - add UserStatsContext (to share user stats state - dateRange and project - between user stats page and certificate page)
  - add stats layout.js for UserStatsContextProvider (applies to `/stats` and `/stats/certificate`)
  - add userStatsDateRange hook (initially state, but added localStorage to persist across browser tabs)
  - add userStatsProject hook (same as useStatsDateRange)
- lib-react-components
  - refactor ZooFooter and ZooHeader with `display: none` when printing (`@media print`)
- lib-user
  - add CertificateContainer and Certificate components for user volunteer certificate display and generating (printing to PDF or printer)
  - add Certificate stories
  - add certificate page to dev app
  - refactor UserStatsContainer for selectedDateRange and selectedProject as props
  - refactor CertificateContainer for selectedDateRange and selectedProject as props 

## How to Review
_Helpful explanations that will make your reviewer happy:_
- What Zooniverse project should my reviewer use to review UX? n/a
- What user actions should my reviewer step through to review this PR?
  - from app-root go to your user stats page (i.e. https://local.zooniverse.org:3000/users/markb-panoptes/stats)
  - confirm user stats page projects and date range selects work as expected
  - click (and right click to new tab) the Generate Volunteer Certificate link to go to your certificate page
  - confirm certificate page reflects user stats project and date range selections
  - click Generate Certificate button/link to open browser print dialog to print certificate to PDF or printer
- Which storybook stories should be reviewed?
  - certificate page: http://localhost:6008/?path=/story/components-certificate--default
- Are there plans for follow up PR’s to further fix this bug or develop this feature? see linked GitHub project

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)
  - Can be used with a screen reader [BBC guide to VoiceOver](https://bbc.github.io/accessibility-news-and-you/assistive-technology/testing-steps/voiceover-mac.html)
  - Can be used from the keyboard [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - It is passing accessibility checks in the storybook

## New Feature
  - Example of SlideTutorial [component](https://github.com/zooniverse/front-end-monorepo/blob/master/packages/lib-classifier/src/components/Classifier/components/SlideTutorial/SlideTutorial.js) with docgen comments, and its [story](https://zooniverse.github.io/front-end-monorepo/@zooniverse/classifier/index.html?path=/docs/other-slidetutorial--default)
